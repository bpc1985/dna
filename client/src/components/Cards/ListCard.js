import React from "react";
import Card from './Card';

import "./ListCard.scss";

export default function ListCard({data, onDetail}) {
  const subscriptionList = data.map((subscription, index) => {
    return (
      <Card
        key={index}
        subscription={subscription}
        onDetail={onDetail}
      />
    );
  });

  return (
    <ul className="subscriptionList">{subscriptionList}</ul>
  );
}