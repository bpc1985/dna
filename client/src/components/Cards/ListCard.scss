.subscriptionList {
  width: 100%;
  margin: 0 auto;
  list-style: none;
  padding: 0;
}