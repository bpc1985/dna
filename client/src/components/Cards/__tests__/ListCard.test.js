import { render } from "@testing-library/react";
import React from "react";
import ListCard from "../ListCard";

const data = [
  {
    _id: "111",
    phone_number: "111111111",
    address: "Address 1",
    dnaPackage: {
      name: "Package 1",
      description: "Description 1",
      type: "mobile"
    }
  },
  {
    _id: "222",
    phone_number: "2222222222",
    address: "Address 2",
    dnaPackage: {
      name: "Package 2",
      description: "Description 2",
      type: "boardband"
    }
  }
];

describe("ListCard", () => {
  it("should render component", () => {
    const onDetail = jest.fn();
    const { container } = render(
      <ListCard data={data} onDetail={onDetail} />
    );
    expect(container).toMatchSnapshot();
  });
});
