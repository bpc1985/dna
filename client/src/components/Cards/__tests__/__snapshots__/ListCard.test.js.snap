// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCard should render component 1`] = `
<div>
  <ul
    class="subscriptionList"
  >
    <li
      class="subscriptionItem"
    >
      <div>
        <h1>
          Package 1
        </h1>
        <h2>
          111111111
        </h2>
      </div>
      <div>
        <span>
          mobile
        </span>
        <button
          class="btn"
        >
          Details
        </button>
      </div>
    </li>
    <li
      class="subscriptionItem"
    >
      <div>
        <h1>
          Package 2
        </h1>
        <h2>
          Address 2
        </h2>
      </div>
      <div>
        <span>
          boardband
        </span>
        <button
          class="btn"
        >
          Details
        </button>
      </div>
    </li>
  </ul>
</div>
`;
